source/extensions/filters/common/expr/library/custom_library.h

```diff
@@ -0,0 +1,82 @@
+#pragma once
+
+#include "envoy/common/pure.h"
+#include "envoy/config/typed_config.h"
+#include "envoy/extensions/rbac/custom_library_config/v3/custom_library.pb.h"
+#include "envoy/protobuf/message_validator.h"
+
+#include "source/common/protobuf/protobuf.h"
+#include "source/extensions/filters/common/expr/context.h"
+#include "source/extensions/filters/common/expr/library/custom_functions.h"
+
+#include "absl/strings/string_view.h"
+#include "eval/public/activation.h"
+#include "eval/public/cel_function_registry.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Filters {
+namespace Common {
+namespace Expr {
+namespace Library {
+
+using envoy::extensions::rbac::custom_library_config::v3::CustomLibraryConfig;
+using google::api::expr::runtime::Activation;
+using google::api::expr::runtime::CelFunctionRegistry;
+
+constexpr absl::string_view CustomVocabularyName = "custom";
+constexpr absl::string_view LazyEvalFuncNameGetDouble = "GetDouble";
+constexpr absl::string_view LazyEvalFuncNameGetProduct = "GetProduct";
+constexpr absl::string_view LazyEvalFuncNameGet99 = "Get99";
+constexpr absl::string_view EagerEvalFuncNameGetNextInt = "GetNextInt";
+constexpr absl::string_view EagerEvalFuncNameGetSquareOf = "GetSquareOf";
+
+class CustomLibrary {
```

Comment:
Slightly related to my comment about adding 'example' to the reference implementation files (custom_vocabulary.h and others) -  I realize now that this class/file is a mix of production code and example implementation code: I think really what we want is the following: (note I'm also suggesting renaming per @RyanTheOptimist's other comment about 'Custom Library' being ambiguous)

Split CustomLibrary into two classes
* 'CustomCELVocabulary' that is a pure interface which defines the methods that custom vocabularies must implement, and is the type used by RBAC and other filters (in the future)
* And 'ExampleCustomCELVocabulary' or similar that inherits from 'CustomCELVocabulary' and exists in its own file(s). This class would be the reference implementation that has member variables etc, and its files would define the play/example functions and their names 'LazyEvalFuncNameGetDouble' etc. We'd rename 'CustomLibraryFactory' to 'ExampleCustomCELVocabularyFactory' as well.

Reply:
ok